soroban-test-wasms/wasm-workspace/create_contract/src/lib.rs

```diff
@@ -7,7 +7,8 @@ pub struct Contract;
 impl Contract {
     // Note that anyone can create a contract here with any salt, so a users call to
     // this could be frontrun and the same salt taken.
-    pub fn create(e: Env, c: Bytes, s: BytesN<32>) {
-        e.create_contract_from_contract(c, s);
+    pub fn create(e: Env, wasm: Bytes, salt: Bytes) {
+        let deployer = e.deployer().from_current_contract(&salt);
```

Comment:
Nitpick: Why not just
```rust
e.deployer().from_current_contract(&salt).deploy(&wasm);
```
I especially don't see any reason to store `_contract_id` on the last line.

Reply:
Fixed. Interestingly wasm had already optimized it out and there was no change to the `.wasm` bytes.